src/mlpack/methods/hmm/hmm.hpp

```diff
@@ -323,14 +323,15 @@ class HMM
               arma::mat& smoothSeq) const;
 
   //! Return the vector of initial state probabilities.
-  const arma::vec& Initial() const { return initial; }
+  const arma::vec Initial() const { return initialProxy; }
   //! Modify the vector of initial state probabilities.
-  arma::vec& Initial() { return initial; }
+  arma::vec& Initial() { recalculateInitial = true; return initialProxy; }
```

Comment:
pedantic style comment, I think that is difficult to read, maybe we should just use two lines here.